### DISCUSSION TITLE
Not getting push token

### DISCUSSION THREAD

**Discussion by Yahya0007** [shinyorg/shiny#988] [Q&A] [answered]
Hi

Firebase push notifications setup.

In Android project 'ShinyApplication.cs' executes 'this.ShinyOnCreate(new StaffApp.StaffAppShinyStartup())' and in 'StaffApp.StaffAppShinyStartup', 'services.UsePush<ShinyPushDelegate>()' gets called. 

However, a Firebase push token never seems to get generated and in ShinyPushDelegate class no method seems to get called; certainly the OnTokenRefreshed(string token) method is not called.

What areas should I look into for this issue?

Thanks

Regards

**Comment by aritchie** [comment, marked as answer]
You don't automatically get a token.  OnTokenRefreshed is ONLY called when the token actually changes which is really only on firebase.  

You need to call pushmanager.RequestAccess() - this will give you what you need (and request/ensure access every startup until you unregister from there)
